pkg/shipper/shipper.go

```diff
@@ -63,6 +70,8 @@ func (s *Shipper) Sync(ctx context.Context) {
 	if err != nil {
 		level.Warn(s.logger).Log("msg", "read dir failed", "err", err)
 	}
+
+	oldestBlockMinTime := int64(0)
```

Comment:
0  is actually a valid timestamp. Since timestamps are `int64` negative values would be as well (in theory).

So the sentinel values we usually work with are `math.MaxInt64` and `math.MinInt64` respectively, whichever makes more sense.